pkg/services/ngalert/ngalert.go

```diff
@@ -99,7 +99,8 @@ func (ng *AlertNG) init() error {
 
 	decryptFn := ng.SecretsService.GetDecryptedValue
 	multiOrgMetrics := ng.Metrics.GetMultiOrgAlertmanagerMetrics()
-	ng.MultiOrgAlertmanager, err = notifier.NewMultiOrgAlertmanager(ng.Cfg, store, store, ng.KVStore, decryptFn, multiOrgMetrics, ng.NotificationService, log.New("ngalert.multiorg.alertmanager"))
+	crypto := notifier.NewCrypto(ng.SecretsService, store)
+	ng.MultiOrgAlertmanager, err = notifier.NewMultiOrgAlertmanager(ng.Cfg, store, store, ng.KVStore, decryptFn, multiOrgMetrics, ng.NotificationService, log.New("ngalert.multiorg.alertmanager"), crypto)
```

Comment:
Now that you've made `Crypto` a slice of the `notifier` because it refers to encryption and decryption of the Alertmanger things. Why not make it public and use it directly? 

You can simplify everything by:

1. Initializing `Crypto` with the Alertmanager itself and making `Crypto` a public method - you can also use delegation if you'd prefer.

```golang
func NewMultiOrgAlertmanager(cfg *setting.Cfg, configStore store.AlertingStore, orgStore store.OrgStore,
	kvStore kvstore.KVStore, decryptFn channels.GetDecryptedValueFn, m *metrics.MultiOrgAlertmanager,
	ns notifications.Service, l log.Logger, c secrets.Service,
) (*MultiOrgAlertmanager, error) {
	moa := &MultiOrgAlertmanager{
		Crypto: NewCrypto(c, configStore),

		logger:        l,
		settings:      cfg,
		alertmanagers: map[int64]*Alertmanager{},
		configStore:   configStore,
		orgStore:      orgStore,
		kvStore:       kvStore,
		decryptFn:     decryptFn,
		metrics:       m,
		ns:            ns,
	}
```

2.  Remove `		Crypto:               crypto` from `*AlertNG`
3. Finally, in `api/api.go` you can do:

```golang
&AlertmanagerSrv{encryption: mam.Crypto, log: logger, ac: api.AccessControl, mam: api.MultiOrgAlertmanager},
```

or 

```golang
// Assuming you prefer delegation.
&AlertmanagerSrv{encryption: api.MultiOrgAlertmanager, log: logger, ac: api.AccessControl, mam: api.MultiOrgAlertmanager},
```